index.js

```diff
@@ -40,31 +40,39 @@ function sameDomain(url) {
 	return location.hostname === a.hostname && location.protocol === a.protocol;
 }
 
-function download(url) {
+function download(url, name) {
 	var a = document.createElement('a');
-	a.download = '';
+	a.download = name || '';
 	a.href = url;
 	// firefox doesn't support `a.click()`...
 	a.dispatchEvent(new MouseEvent('click'));
 }
 
-module.exports = function (urls) {
+module.exports = function (urls, renameFn) {
 	if (!urls) {
 		throw new Error('`urls` required');
 	}
+	if (!renameFn) {
+		renameFn = function () { return ''; };
+	}
 
 	if (typeof document.createElement('a').download === 'undefined') {
 		return fallback(urls);
 	}
 
 	var delay = 0;
 
-	urls.forEach(function (url) {
+	urls.map(function (url, index, urls) {
```

Comment:
Instead of mapping through the `urls`, why don't you set the name inside the `forEach` or in the `download` function? Then we don't have to do a unnecessary loop.

```js
var name = typeof renameFn === 'function' ? renameFn(url, index, urls) : '';
```